src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
+               ConnectionInfo connInfo;
+               try
+               {
+                   if (connectionService.TryFindConnection(
+                           requestParams.OwnerUri,
+                           out connInfo))
+                   {
+                       using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
+                       {
+                           IRenamable renameObject = this.GetSQLRenameObject(requestParams, sqlConn);
+
+                           renameObject.Rename(requestParams.NewName);
+                       }
+                   }
+                   else
+                   {
+                       Logger.Error("The connection could not be found.");
+                       throw new Exception(SR.ErrorConnectionNotFound);
+                   }
+               }
+               catch (Exception e)
+               {
+                   Logger.Error($"An error occurred while renaming {requestParams.UrnOfObject}: " + e);
+                   throw new Exception(SR.ErrorOnRenameOperationMessage, e);
+               }
+               await requestContext.SendResult(true);
+           });
+        }
+
+        /// <summary>
+        /// Method to get the sql object, which should be renamed
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="connection">the sqlconnection on the server to search for the sqlobject</param>
+        /// <returns>the sql object if implements the interface IRenamable, so they can be renamed</returns>
+        private IRenamable GetSQLRenameObject(RenameRequestParams requestParams, SqlConnection connection)
+        {
+            ServerConnection serverConnection = new ServerConnection(connection);
+            Server server = new Server(serverConnection);
+            SqlSmoObject dbObject = server.GetSmoObject(new Urn(requestParams.UrnOfObject));
```

Comment:
>GetSmoO [](http://example.com/codeflow?start=43&length=7)

does this throw if it can't find the object, or does it return null? if it returns null, you should check for it in the caller and return a specific error about the object not being found. Otherwise the exception message is going to be for a null reference

Reply:
Looks like it throws 

https://github.com/microsoft/sqlmanagementobjects/blob/main/src/Microsoft/SqlServer/Management/Smo/serverbase.cs#L668

Reply:
I do not think, that null will be returned (As i understand it, it throws an [MissingObjectException](https://github.com/microsoft/sqlmanagementobjects/blob/04249a2dc85ad1aa0d492b8a2cecf65583ded177/src/Microsoft/SqlServer/Management/Smo/serverbase.cs#L666)). I also tested for Object, which [do not exist](https://github.com/M-Patrone/sqltoolsservice/blob/feature-rename-sql-objects/test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/ObjectManagementServiceTests.cs#L82). Should I add it nevertheless?

Reply:
since SMO throws, no need to check for null. thx